Makefile

```diff
@@ -250,3 +250,66 @@ release-manifests: $(KUSTOMIZE) ## Create kustomized release manifest in $RELEAS
 release-templates: ## Generate release templates
 	@mkdir -p $(RELEASE_DIR)
 	cp templates/cluster-template*.yaml $(RELEASE_DIR)/
+
+
+##@ e2e
+## --------------------------------------
+## E2e tests
+## --------------------------------------
+ROOT_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+E2E_CONF_FILE ?= $(ROOT_DIR)/test/e2e/config/proxmox-ci.yaml
+E2E_CONF_FILE_ENVSUBST := $(ROOT_DIR)/test/e2e/config/proxmox-ci-envsubst.yaml
+E2E_DATA_DIR ?= $(ROOT_DIR)/test/e2e/data
+KUBETEST_CONF_PATH ?= $(abspath $(E2E_DATA_DIR)/kubetest/conformance.yaml)
+
+
+# Allow overriding the e2e configurations
+GINKGO_FOCUS ?= Workload cluster creation
+GINKGO_SKIP ?= API Version Upgrade
+GINKGO_NODES ?= 1
+GINKGO_NOCOLOR ?= false
+GINKGO_ARGS ?=
+GINKGO_TIMEOUT ?= 2h
+GINKGO_POLL_PROGRESS_AFTER ?= 10m
+GINKGO_POLL_PROGRESS_INTERVAL ?= 1m
+ARTIFACTS ?= $(ROOT_DIR)/_artifacts
+SKIP_CLEANUP ?= false
+SKIP_CREATE_MGMT_CLUSTER ?= false
+
+# Install tools
+
+GINKGO_BIN := ginkgo
+GINKGO := $(LOCALBIN)/$(GINKGO_BIN)
+GINKGO_PKG := $(shell sed '/github.com\/onsi\/ginkgo\/v2/!d; s/ /\/ginkgo@/' go.mod)
+
+ENVSUBST_VER := v1.4.2
+ENVSUBST_BIN := envsubst
+ENVSUBST := $(LOCALBIN)/$(ENVSUBST_BIN)
+
+$(ENVSUBST): ## Build envsubst.
+	test -s $(LOCALBIN)/$(ENVSUBST_BIN) || \
+	GOBIN=$(LOCALBIN) go install github.com/a8m/envsubst/cmd/envsubst@$(ENVSUBST_VER)
+
+$(GINKGO): ## Build ginkgo.
+	GOBIN=$(LOCALBIN) go install $(GINKGO_PKG)
```

Comment:
Just use `github.com/onsi/ginkgo/v2/ginkgo`. The version is fetched from go.mod if it's not provided

Reply:
Wiktor already pushed that change.

Reply:
Good to know, so can we just delete the version checking dance altogether, then?

Reply:
If you're referring to the controller-gen and kustomize version checking, then not quite. Unless you reinstall everything on each make invocation, you must somehow check that the local binary has the correct version.
`go install github.com/onsi/ginkgo/v2/ginkgo` will just ensure that the version it uses matches go.mod. Update need to be handled separately.

Reply:
At least get rid of the `GINKGO_PKG` variable

Reply:
and replace it with what?